src/DirectionProvider.jsx

```diff
@@ -45,10 +46,10 @@ export default class DirectionProvider extends React.Component {
   }
 
   render() {
-    const { children, direction, inline } = this.props;
+    const { children, direction, inline, lang } = this.props;
     const Tag = inline ? 'span' : 'div';
     return (
-      <Tag dir={direction}>
+      <Tag dir={direction} lang={lang}>
```

Comment:
```suggestion
      <Tag dir={direction} ...(lang && { lang })>
```
that way there's no `lang` prop when `lang` isn't provided?

Reply:
I don't think this actually makes any difference. See here:

![image](https://user-images.githubusercontent.com/195534/135451329-2cd4ced9-a407-4d1b-82a5-0abde7286e85.png)

```jsx
function Bar(props) {
  return <div>{JSON.stringify(props)}</div>;
}

function Foo(props) {
  return <Bar dir={props.dir} lang={props.lang} />;
}

ReactDOM.render(
  <Foo dir="rtl" />,
  document.getElementById('hello-example')
);
```

Reply:
That's because `JSON.stringify` specially removes `undefined` values. The prop is still actually present, so any "exact" propTypes, TS types, or Flow types would error on it.

Reply:
Fair enough! I've added that